src/components/AdvancedSearch/AdvancedSearch.tsx

```diff
@@ -83,183 +133,176 @@ const AdvancedSearch: React.FC<{
       return query.field !== queryKey;
     });
-
     // If the new query is not empty, add it
     if (newQuery.query.length > 0) {
       allQueries.push(newQuery);
     }
-    setSearchQuery({
-      ...searchQuery,
-      queries: allQueries,
-    });
+    setQueries([...allQueries]);
   };
 
   const onLanguageChange = (e, language) => {
-    const languageFilters = findFiltersForField(
-      searchQuery.filters,
-      filterFields.language
-    );
-
-    setSearchQuery({
-      ...searchQuery,
-      filters: [
-        ...findFiltersExceptField(searchQuery.filters, filterFields.language),
-        ...(e.target.checked
-          ? [
-              ...languageFilters,
-              { field: filterFields.language, value: language },
-            ]
-          : languageFilters.filter((filter) => {
-              return filter.value !== language;
-            })),
-      ],
-    });
+    setLanguageFilters([
+      ...(e.target.checked
+        ? [
+            ...languageFilters,
+            { field: filterFields.language, value: language },
+          ]
+        : languageFilters.filter((filter) => {
+            return filter.value !== language;
+          })),
+    ]);
   };
 
   const onBookFormatChange = (e, format) => {
-    const formatFilters = findFiltersForField(
-      searchQuery.filters,
-      filterFields.format
-    );
+    const newFilters = [
+      ...(e.target.checked
+        ? [...formatFilters, { field: filterFields.format, value: format }]
+        : formatFilters.filter((filter) => {
+            return filter.value !== format;
+          })),
+    ];
 
-    setSearchQuery({
-      ...searchQuery,
-      filters: [
-        ...findFiltersExceptField(searchQuery.filters, filterFields.format),
-        ...(e.target.checked
-          ? [...formatFilters, { field: "format", value: format }]
-          : formatFilters.filter((filter) => {
-              return filter.value !== format;
-            })),
-      ],
-    });
+    setFormatFilters(newFilters);
   };
 
   const onDateChange = (
-    e: React.FormEvent<HTMLInputElement>,
+    e: React.ChangeEvent<HTMLInputElement>,
     isStart: boolean
   ) => {
     const field = isStart ? filterFields.startYear : filterFields.endYear;
-    const newFilters = [
-      ...findFiltersExceptField(searchQuery.filters, field),
-      ...[{ field: field, value: e.currentTarget.value }],
-    ];
-    setSearchQuery({
-      ...searchQuery,
-      filters: newFilters,
-    });
+    const newFilter = {
+      field: field,
+      value: e.currentTarget.value,
+    };
+    if (isStart) setStartFilter(newFilter);
+    else setEndFilter(newFilter);
   };
 
-  const startFilter = findFiltersForField(
-    searchQuery.filters,
-    filterFields.startYear
-  );
-  const endFilter = findFiltersForField(
-    searchQuery.filters,
-    filterFields.endYear
+  // Because each FormRow has two InputTerms each,
+  // create an array of nested arrays, each with two InputTerms.
```

Comment:
This is completely personal preference, but because there aren't that many terms and there aren't plans to add more, I personally prefer to just write out the markup rather than doing a double loop through a double array.